beginner_source/dist_overview.rst

```diff
@@ -15,107 +15,62 @@ to the technology that can best serve your use case.
+
+DTensor and DeviceMesh are primitives used to build parallelism in terms of sharded or replicated tensors on N-dimensional process groups.
+* `DTensor <https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/README.md>`__ represents a tensor that is sharded and/or replicated, and communicates automatically to reshard tensors as needed by operations.
+* `DeviceMesh <https://pytorch.org/docs/stable/distributed.html#devicemesh>`__ abstracts multiple dimensions of parallelism and manages the underlying ProcessGroups used for collective communications,
+
+
+`C10D <https://pytorch.org/docs/stable/distributed.html>`__
   offers both collective communication APIs (e.g.,
   `all_reduce <https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_reduce>`__
   and `all_gather <https://pytorch.org/docs/stable/distributed.html#torch.distributed.all_gather>`__)
   and P2P communication APIs (e.g.,
   `send <https://pytorch.org/docs/stable/distributed.html#torch.distributed.send>`__
-  and `isend <https://pytorch.org/docs/stable/distributed.html#torch.distributed.isend>`__).
-  DDP and RPC (`ProcessGroup Backend <https://pytorch.org/docs/stable/rpc.html#process-group-backend>`__)
-  are built on c10d, where the former uses collective communications
-  and the latter uses P2P communications. Usually, developers do not need to
-  directly use this raw communication API, as the DDP and RPC APIs can serve
-  many distributed training scenarios. However, there are use cases where this API
-  is still helpful. One example would be distributed parameter averaging, where
-  applications would like to compute the average values of all model parameters
-  after the backward pass instead of using DDP to communicate gradients. This can
-  decouple communications from computations and allow finer-grain control over
-  what to communicate, but on the other hand, it also gives up the performance
-  optimizations offered by DDP.
+  and `isend <https://pytorch.org/docs/stable/distributed.html#torch.distributed.isend>`__),
+  which are used under the hood in all of the parallelism implementations.
   `Writing Distributed Applications with PyTorch <../intermediate/dist_tuto.html>`__
   shows examples of using c10d communication APIs.
 
 
 Data Parallel Training
 ----------------------
 
+Data Parallelism is a widely adopted single-program multiple-data training paradigm
+where the model is replicated on every process, every model replica computes local gradients for
+a different set of input data samples, gradients are globally averaged before each optimizer step.
+
 PyTorch provides several options for data-parallel training. For applications
 that gradually grow from simple to complex and from prototype to production, the
 common development trajectory would be:
 
 1. Use single-device training if the data and model can fit in one GPU, and
    training speed is not a concern.
 2. Use single-machine multi-GPU
```

Comment:
Shall we unify `Use single-machine multi-GPU` and `Use multi-machine`? IMO the previous doc is to distinguish the old DP and the DDP (DP is single process I think so there were some difference). But now for pytorch distributed we can universally recommend "multi-gpu" DDP